### PR TITLE
build custom metadata from config

### PR DESCRIPTION
**Pull request by jessopb**
Used webpack transform to insert custom metadata into html on compile.

### CODE REVIEW

File: web/.env.defaults [hunk 1]
```diff
@@ -5,6 +5,8 @@ WEB_SERVER_PORT=1337
 DOMAIN=lbry.tv
```
Comment (seanyesmunt) [inline web/.env.defaults:5]: What is the purpose of having two of these files?

Comment (jessopb) [reply web/.env.defaults:5]: The project structure means root for web build is different from root for electron build.

Comment (jessopb) [reply web/.env.defaults:5]: can you check in a symlink? maybe a webpack pre-copy?

File: web/src/html.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ const path = require('path');
 
 let html = fs.readFileSync(path.join(__dirname, '/../dist/index.html'), 'utf8');
 
+module.exports = { insertToHead, buildBasicOgMetadata, getHtml };
```
Comment (seanyesmunt) [inline web/src/html.js:11]: Move this to the very bottom of the file

File: web/src/html.js [hunk 3]
```diff
@@ -38,15 +39,35 @@ function buildOgMetadata(overrideOptions = {}) {
...
+    `<meta name="twitter:description" content="${description || SITE_DESCRIPTION}" />\n` +
+    `<meta name="twitter:image" content="${URL}/public/v2-og.png"/>\n` +
+    `<meta name="twitter:url" content="${URL}" />\n` +
     '<meta property="fb:app_id" content="1673146449633983" />';
+  return head;
+}
 
+function buildBasicOgMetadata() {
+  const head =
+    '    <!-- VARIABLE_HEAD_BEGIN -->' +
```
Comment (seanyesmunt) [inline web/src/html.js:57]: I think this should be removed from the HTML file (but keep the comment?) now that this exists?
https://github.com/lbryio/lbry-desktop/blob/master/static/index-web.html#L66-L74

File: web/src/html.js [hunk 4]
```diff
@@ -38,15 +39,35 @@ function buildOgMetadata(overrideOptions = {}) {
...
+    '<title>lbry.tv</title>\n' +
+    `<meta property="og:url" content="${URL}" />\n` +
+    `<meta property="og:title" content="${SITE_TITLE}" />\n` +
+    `<meta property="og:site_name" content="${SITE_TITLE} | ${SITE_MOTTO}"/>\n` +
+    `<meta property="og:description" content="${SITE_DESCRIPTION}" />\n` +
+    `<meta property="og:image" content="${URL}/public/v2-og.png" />\n` +
+    '<meta name="twitter:card" content="summary_large_image"/>\n' +
+    `<meta name="twitter:title" content="${SITE_TITLE}" />\n` +
+    `<meta name="twitter:description" content="${SITE_DESCRIPTION}" />\n` +
+    `<meta name="twitter:image" content="${URL}/public/v2-og.png"/>\n` +
```
Comment (seanyesmunt) [inline web/src/html.js:67]: This is pretty much the same as the function above. Can they just be combined?

**Comment by seanyesmunt** [comment]
Looks good. Is there any chance this could break web/lat when this is merged?

File: .env.defaults [hunk 5]
```diff
@@ -1,14 +1,30 @@
...
-SITE_TITLE=lbry.tv
-LOGO_TITLE=LBRY
 LBRY_WEB_API=https://api.lbry.tv
 LBRY_WEB_STREAMING_API=https://cdn.lbryplayer.xyz
 WELCOME_VERSION=1.0
+
+# Custom Site info
+DOMAIN=lbry.tv
+URL=https://lbry.tv
+SITE_TITLE=lbry.tv
```
Comment (kauffj) [inline .env.defaults:16]: HOMEPAGE_TITLE and DEFAULT_TITLE ?

Comment (kauffj) [reply .env.defaults:16]: And can we add TITLE_SUFFIX?

Comment (kauffj) [reply .env.defaults:16]: OG_IMAGE

**Comment by jessopb** [comment]
Provides two ways to have custom og-image: webpack copies og-v2.png from custom, or set OG_IMAGE_URL in .env.
Adds rel=canonical config

**Comment by jessopb** [comment]
Must modify .env on prod (for matomo id) and lbrylat

**Comment by seanyesmunt** [comment]
https://lbry.tv/shipit

**Pull request merged**